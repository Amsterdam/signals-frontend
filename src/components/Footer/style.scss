.footer-component {
  padding: 20px;
}
