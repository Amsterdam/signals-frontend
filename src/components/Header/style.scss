@import '~stijl/dist/scss/ams-colorpalette';
@import '~stijl/dist/scss/ams-breakpoints';

.header-component {
  .mainlogo {
    text-decoration: none;

    .logoset .screen-logo {
      width: 20px;
    }
  }

  .links {
    margin-top: 1.125rem;

    // TODO: remove this li {} block and uncomment the one below it once npm package stijl v2.9.2 available

    li {
      text-align: right;
    }
      /*
      li {
        text-align: left;

        @include media-breakpoint-up(sm) {
          text-align: right;
        }
      }
      */
  }
}
