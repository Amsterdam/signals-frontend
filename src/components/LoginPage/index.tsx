// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2019 - 2021 Gemeente Amsterdam
import styled from 'styled-components'
import {
  Row,
  Column,
  Paragraph,
  themeColor,
  themeSpacing,
} from '@amsterdam/asc-ui'

import Button from 'components/Button'
import ButtonBar from 'components/ButtonBar'
import { useDispatch } from 'react-redux'
import { doLogin } from 'containers/App/actions'

const Notification = styled.div`
  border-left: 3px solid ${themeColor('secondary')};
  margin: ${themeSpacing(6)} 0;
  padding-left: ${themeSpacing(5)};
`

const LoginPage = () => {
  const dispatch = useDispatch()

  const handleLogin = () => {
    dispatch(doLogin())
  }

  return (
    <Row data-testid="loginPage">
      <Column span={12}>
        <Notification>
          <Paragraph>
            Om deze pagina te zien dient u ingelogd te zijn.
          </Paragraph>

          <ButtonBar>
            <Button
              variant="secondary"
              data-testid="loginButton"
              onClick={handleLogin}
              type="button"
            >
              <span>Inloggen</span>
            </Button>
          </ButtonBar>
        </Notification>
      </Column>
    </Row>
  )
}

export default LoginPage
