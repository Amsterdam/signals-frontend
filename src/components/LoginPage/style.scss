.login-page {
  margin-top: 150px;
}
