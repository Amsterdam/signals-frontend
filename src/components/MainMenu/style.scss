@import '~amsterdam-stijl/dist/scss/ams-colorpalette';

.main-menu-component {
  background-color: $ams-top-menu;
}
