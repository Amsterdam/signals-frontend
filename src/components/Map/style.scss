.map-component {
  #mapdiv {
    position: relative;
    width: 100%;
    height: 450px;
  }

  &__loading {
    background-color: rgba(0, 0, 0, .2);
    display: block;
    width: calc(100% - 30px); // padding left and right of grid
    height: 100%;
    position: absolute;
    z-index: 1001;
  }
}
