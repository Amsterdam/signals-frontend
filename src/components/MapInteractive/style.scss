.map-component {
  margin-top: 5px;

  #mapdiv {
    position: relative;
    width: 100%;
    height: 450px;
  }

  .embed-search {
    // pull the search input outside of the map
    // top: -50px;
    // left: 0;
    max-width: 100%;
    // width: 100%;

    input[type="text"] {
      height: inherit;
    }

    button {
      display: none;
    }

    &__auto-suggest {
      width: 100%;
    }

    .invoer {
      box-shadow: none;
    }
  }

  &__loading {
    background-color: rgba(0, 0, 0, .2);
    display: block;
    width: calc(100% - 30px); // padding left and right of grid
    height: 100%;
    position: absolute;
    z-index: 1001;
  }
}
