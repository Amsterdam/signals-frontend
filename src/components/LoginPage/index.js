import React from 'react';

import { login } from 'shared/services/auth/auth';

import './style.scss';


const LoginPage = () => (
  <div className="login-page notification notification-red margin-top-bottom">
    <div className="col-12">
      <p>
        Om deze pagina te zien dient u ingelogd te zijn.
      </p>
      <button className="action primary" onClick={() => { login('datapunt'); }}>
        <span className="value">Inloggen</span>
      </button>
      <button className="action primary" onClick={() => { login('grip'); }}>
        <span className="value">Inloggen ADW</span>
      </button>
    </div>
  </div>
);

export default LoginPage;
