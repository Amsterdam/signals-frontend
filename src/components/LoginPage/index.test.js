import React from 'react';
import { shallow } from 'enzyme';

import { login } from 'shared/services/auth/auth';
import LoginPage from './index';
import './style.scss';

jest.mock('shared/services/auth/auth');

describe('<LoginPage />', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = shallow(<LoginPage />);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should login on datapunt when Inloggen button is clicked', () => {
    renderedComponent.find('button').at(0).simulate('click');
    expect(login).toHaveBeenCalledWith('datapunt');
  });

  it('should login on datapunt when Inloggen ADW button is clicked', () => {
    renderedComponent.find('button').at(1).simulate('click');
    expect(login).toHaveBeenCalledWith('grip');
  });
});
