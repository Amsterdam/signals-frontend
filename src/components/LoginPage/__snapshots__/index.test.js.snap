// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render correctly 1`] = `
<div
  className="login-page notification notification-red margin-top-bottom"
>
  <div
    className="col-12"
  >
    <p>
      Om deze pagina te zien dient u ingelogd te zijn.
    </p>
    <button
      className="action primary"
      onClick={[Function]}
    >
      <span
        className="value"
      >
        Inloggen
      </span>
    </button>
    <button
      className="action primary"
      onClick={[Function]}
    >
      <span
        className="value"
      >
        Inloggen ADW
      </span>
    </button>
  </div>
</div>
`;
